ddtrace/tracer/abandonedspans.go

```diff
@@ -0,0 +1,305 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"container/list"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var (
+	tickerInterval = 1 * time.Minute
+	logSize        = 9000
+)
+
+// bucket is a generic implementation of a dynamic collection of elements
+// stored under a value-bound key (like time). Inspired by concentrator.rawBucket.
```

Comment:
Let's say explicitly that the implementation is not thread-safe. It might be useful for future users of this struct.